src/parseTools.js

```diff
@@ -920,6 +925,36 @@ function to64(x) {
   return `BigInt(${x})`;
 }
 
+// Generates an expression that tests whether a pointer (either a BigInt or a signed int32 JS Number) is aligned to the given byte multiple.
+function isPtrAligned(ptr, alignment) {
```

Comment:
Do you need to second argument here?  Shouldn't pointer always be aligned to pointer size?  i.e. should alignment here not always be POINTER_SIZE?

Reply:
This method is used in all the places where we previously did something like this

`assert((value & 3) == 0, 'Pointer to integer data passed to glUniform2iv must be aligned to four bytes!');`

So for example, if we want to pass int32 values to webGL (e.g. via glUniform2iv) then this data has to be aligned to 4 bytes because it is being passed via the HEAP32 array buffer view. To be honest, when i first saw this code i was a bit weary, since on the C/C++ side such guarantees are never required and i am pretty sure, that i could write some simple code which would break this assumption. But as far as i can tell, up until now, our production code has never caused any issues here.

As for whether the second parameter is required. I think currently we are always either using 4 byte alignment or 1 byte alignment (i.e. unaligned). So there is definitely space for improvement. Will take a closer look at this.